### PR TITLE
Controllers for Orders

### PR DESCRIPTION
**Pull request by ghost**
Added Controllers and Models for Order.

### CODE REVIEW

File: internal/controller/order.go [hunk 1]
```diff
@@ -0,0 +1,55 @@
+package controller
+
+import (
+	"Tradeasy/internal/model/transaction_engine"
+	"Tradeasy/internal/services"
+	"github.com/gin-gonic/gin"
+	"net/http"
+)
+func AbortMsg(code int, err error, c *gin.Context) {
```
Comment (mukesh0513) [inline internal/controller/order.go]: what is AbortMsg?

File: internal/model/transaction_engine/stocksfeed.go [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+	Open      int                 `json:"open" gorm:"column:open"`
+	High      int                 `json:"high" gorm:"column:high"`
+	Low       int                 `json:"low" gorm:"column:low"`
+	TradedAt  timestamp.Timestamp `json:"traded_at" gorm:"column:traded_at"`
+	CreatedAt timestamp.Timestamp `json:"created_at" gorm:"column:created_at"`
+	DeletedAt timestamp.Timestamp `json:"deleted_at" gorm:"column:deleted-at"`
+}
+
+func (s *StocksFeed) TableName() string {
+	return "StocksFeed"
```
Comment (mukesh0513) [inline internal/model/transaction_engine/stocksfeed.go]: snake_case

File: internal/model/transaction_engine/order_history.go [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+	BuyPrice      int                 `json:"buy_price" gorm:"column:buy_price"`
+	SellPrice     int                 `json:"sell_price" gorm:"column:sell_price"`
+	CommissionFee int                 `json:"commission_fee" gorm:"column:commission_fee"`
+	BoughtAt      timestamp.Timestamp `json:"bought_at" gorm:"column:bought_at"`
+	SoldAt        timestamp.Timestamp `json:"sold_at" gorm:"column:sold_at"`
+	CreatedAt     timestamp.Timestamp `json:"created_at" gorm:"column:created_at"`
+	UpdatedAt     timestamp.Timestamp `json:"updated_at" gorm:"column:updated_at"`
+	DeletedAt timestamp.Timestamp `json:"deleted_at" gorm:"column:deleted_at"`
+}
+
```
Comment (mukesh0513) [inline internal/model/transaction_engine/order_history.go]: run gofmt

**Comment by mukesh0513** [comment]
Better pr names from next time.

**Comment by mukesh0513** [comment]
run go fmt.

File: internal/model/holdings.go [hunk 4]
```diff
@@ -0,0 +1,20 @@
+package model
+
+import "github.com/golang/protobuf/ptypes/timestamp"
+
+type Holdings struct{
+	Userid        string			  `gorm:"foreign_key; column:User" json:"userid"`
```
Comment (mukesh0513) [inline internal/model/holdings.go]: camelCase.

File: internal/model/stocksfeed.go [hunk 5]
```diff
@@ -0,0 +1,22 @@
+package model
```
Comment (mukesh0513) [inline internal/model/stocksfeed.go]: file name snake_case.

File: internal/model/stocksfeed.go [hunk 6]
```diff
@@ -0,0 +1,22 @@
...
+)
+
+type StocksFeed struct {
+	Id        big.Int             `gorm:"primary_key;column:id" json:"id"`
+	StockName string              `gorm:"column:stock_name" json:"stock_name" `
+	LTP       int                 `gorm:"column:ltp" json:"ltp" `
+	Open      int                 `gorm:"column:open" json:"open" `
+	High      int                 `gorm:"column:high" json:"high" `
+	Low       int                 `gorm:"column:low" json:"low" `
+	TradedAt  timestamp.Timestamp `gorm:"column:traded_at" json:"traded_at" `
```
Comment (mukesh0513) [inline internal/model/stocksfeed.go]: follow the same convention for timestamp.

Comment (ghost) [reply internal/model/stocksfeed.go]: I have used timestamp.Timestamp in all my files

File: internal/controller/order.go [hunk 7]
```diff
@@ -0,0 +1,51 @@
...
+	"github.com/gin-gonic/gin"
+	"net/http"
+)
+
+func BuyOrder(c *gin.Context) {
+	var BReq Order.BuyRequest
+	id := c.Params.ByName("Userid")
+	c.BindJSON(&BReq)
+	BReq.UserId = id
+	err, bres := Order.BuyOrder(BReq)
```
Comment (mukesh0513) [inline internal/controller/order.go]: follow (response, error) convention.

File: internal/controller/order.go [hunk 8]
```diff
@@ -0,0 +1,51 @@
...
+
+import (
+	"Tradeasy/internal/services/Order"
+	"fmt"
+	"github.com/gin-gonic/gin"
+	"net/http"
+)
+
+func BuyOrder(c *gin.Context) {
+	var BReq Order.BuyRequest
```
Comment (mukesh0513) [inline internal/controller/order.go]: variable name camelCase and same comment for other files.

File: internal/services/Order/order.go [hunk 9]
```diff
@@ -0,0 +1,13 @@
+package Order
```
Comment (mukesh0513) [inline internal/services/Order/order.go]: filename small case.

File: internal/services/Order/order_body.go [hunk 10]
```diff
@@ -0,0 +1,47 @@
+package Order
```
Comment (mukesh0513) [inline internal/services/Order/order_body.go]: reword filename. order_body is vague.

**Review by mukesh0513 — APPROVED** [review]
LGTM

**Pull request merged**